files/KoreBuild/scripts/KoreBuild.psm1

```diff
@@ -111,25 +101,145 @@ function Invoke-RepositoryBuild(
+    {
+        $infoOutput = dotnet --version
+
+        $version = $infoOutput.SubString(0, $infoOutput.IndexOf('.'))
+        $versionInt = [convert]::ToInt32($version, 10)
+        Write-Host "Dotnet version was $versionInt"
+        return $versionInt
+    }
+    else {
+        return 0
+    }
+}
+
+<#
+.SYNOPSIS
+Set the settings.
+
+.DESCRIPTION
+Set the settings which will be used by other commands
 
 .PARAMETER ToolsSource
 The base url where build tools can be downloaded.
 
 .PARAMETER DotNetHome
 The directory where tools will be stored on the local machine.
+
+.PARAMETER RepoPath
+The directory to execute the command against.
 #>
-function Install-Tools(
-    [Parameter(Mandatory = $true)]
+function Set-KoreBuildSettings(
     [string]$ToolsSource,
-    [Parameter(Mandatory = $true)]
-    [string]$DotNetHome) {
+    [string]$DotNetHome,
+    [string]$RepoPath
+)
+{
+    if (!$DotNetHome) {
+        $DotNetHome = if ($env:DOTNET_HOME) { $env:DOTNET_HOME } `
+            elseif ($env:USERPROFILE) { Join-Path $env:USERPROFILE '.dotnet'} `
+            elseif ($env:HOME) {Join-Path $env:HOME '.dotnet'}`
+            else { Join-Path $PSScriptRoot '.dotnet'}
+    }
+
+    if (!$ToolsSource) { $ToolsSource = 'https://aspnetcore.blob.core.windows.net/buildtools' }
+
+    $global:KoreBuildSettings = @{
+        ToolsSource = $ToolsSource
+        DotNetHome = $DotNetHome
+        RepoPath = $RepoPath
+    }
+}
+
+<#
+.SYNOPSIS
+Execute the given command.
 
+.PARAMETER Command
+The command to be executed.
+
+.PARAMETER Arguments
+Arguments to be passed to the command.
+
+.EXAMPLE
+Invoke-KoreBuildCommand "docker-build" /t:Package
+
+.NOTES
+Set-KoreBuildSettings must be called before Invoke-KoreBuildCommand.
+#>
+function Invoke-KoreBuildCommand(
+    [Parameter(Mandatory=$true)]
+    [string]$Command,
+    [Parameter(ValueFromRemainingArguments = $true)]
+    [string[]]$Arguments
+)
+{
+    if(!(Get-Variable KoreBuildSettings -Scope Global -ErrorAction SilentlyContinue))
+    {
+        throw "Set-KoreBuildSettings must be called before Invoke-KoreBuildCommand."
+    }
+
+    if($Command -eq "default-build")
+    {
+        Install-Tools
+        Invoke-RepositoryBuild @Arguments
+    }
+    elseif($Command -eq "msbuild")
+    {
+        Invoke-RepositoryBuild @Arguments
+    }
+    elseif ($Command -eq "install-tools") {
+        Install-Tools
+    }
+    else {
+        Ensure-Dotnet
+
+        $korebuildConsoleDll = Get-KoreBuildConsole
+
+        & dotnet $korebuildConsoleDll $Command `
+            --tools-source $global:KoreBuildSettings.ToolsSource `
+            --dotnet-home $global:KoreBuildSettings.DotNetHome `
+            --repo-path $global:KoreBuildSettings.RepoPath `
+            $Arguments
+    }
+}
+
+function Get-KoreBuildConsole()
+{
+    return Join-Paths $PSScriptRoot ("..", "tools", "KoreBuild.Console.dll")
+}
+
+<#
+.SYNOPSIS
+Installs tools if required.
+#>
+function Install-Tools()
```

Comment:
Same here. Keep `$DotNetHome` as a parameter and fallback to `$global:KoreBuildSettings`.